lnst/Agent/conditions/WaitForConditionModule.py

```diff
@@ -0,0 +1,49 @@
+import time
+import signal
+import logging
+from abc import abstractmethod
+
+from lnst.Common.BaseModule import BaseModule
+
+
+class WaitForConditionModule(BaseModule):
+    def __init__(self, timeout: int = 30, **kwargs):
+        super().__init__(**kwargs)
+
+        self._timeout = timeout
+
+    @property
+    def timeout(self):
+        return self._timeout
+
+    def run(self):
+        logging.info(f"Waiting for condition {self.__class__.__name__} to be met")
+        signal.signal(signal.SIGALRM, self._sig_handler)
+        signal.alarm(self._timeout)
+
+        counter = 0
+        try:
+            while not self._condition():
+                time.sleep(1)
+                logging.debug(f"Waiting for condition: {counter}/{self._timeout}")
+                counter += 1
+        except TimeoutError:
+            logging.exception("Timeout of conditional wait reached!")
+            return False
+        finally:
+            signal.alarm(0)
+            logging.info(f"Condition {self.__class__.__name__} met/timeouted")
+
+        return True
+
+    @staticmethod
+    def _sig_handler(signum, _):
+        if signum == signal.SIGALRM:
+            raise TimeoutError("Timeout reached")
+
+        return
+    
+    @abstractmethod
+    def _condition(self):
```

Comment:
i'm thinking if maybe we should have a default implementation of this condition that is provided via an init parameter?

```
host.wait_for_condition(WaitForConditionModule(condition=some_function))
```

you could still override this via a derived class, but you could also use this for simple lambda function conditions without having to define an extra class for them.

Reply:
The problem of this approach is, that the condition needs to be pickled and sent to the host. 

However, python's `pickle` module doesn't really pickles whole classes/functions, just a reference with instance data therefore we can't use `lamba` functions. Functions are pickled as a `module_path.function_name` so lambda functions cannot be pickled, as they don't have the name.

Support for "function condition" wouldn't be a problem. However, if the condition function is located in the same file as the recipe is, recipe module is then loaded on the host side which caused a weird cyclic import error. Spent some time trying to make it work but then I decided to simply move conditions to separate class and module which solves both of the problems.

There is also python library called `dill` which transfers whole class+instance data or lambda functions but that would make this MR even larger.

Reply:
agreed with the lambda part, discussed the named function a bit more in person, i'm interested in the generated errors to check if there isn't a simple solution.